htsget-http/src/http_core.rs

```diff
@@ -19,7 +17,7 @@ use crate::{
 /// consulted [here](https://samtools.github.io/hts-specs/htsget.html)
 #[instrument(level = "debug", skip_all, ret)]
 pub async fn get(
-  searcher: Arc<impl HtsGet + Send + Sync + 'static>,
```

Comment:
Nice! Getting rid of Arc mean introducing Clone, but I guess the structs are not that big anyway, so I suspect this is a win?

Reply:
It's not really cloning too much anyway. I think the only spot it happens is when it receives multiple `POST` requests which can be processed in parallel.

But yes, the structs are pretty shallow. They basically only hold some `Box` pointers to the heap, which is more or less what `Arc` does anyway, so it should be pretty similar in terms of performance.